root/etc/systemd/system/discovery-setup.service

```diff
@@ -1,13 +1,10 @@
 [Unit]
 Description=Download zip extensions to this image
-Wants=basic.target
-After=basic.target network.target network-online.target nss-lookup.target
+Wants=discovery-autostart.service
+Before=discovery-autostart.service
 
 [Service]
 Type=oneshot
-EnvironmentFile=-/etc/default/discovery-zip-server
+EnvironmentFile=/etc/default/discovery-zip-server
 RemainAfterExit=yes
 ExecStart=/usr/bin/discovery-setup
-
-[Install]
-WantedBy=multi-user.target
```

Comment:
To be honest, I was never fan of `discovery-setup`. It's so misleading, I'd prefer `discovery-fetch-extensions` or something... ;-)